src/lib/containers/UserCard.tsx

```diff
@@ -62,48 +63,67 @@ export const UserCard: React.FunctionComponent<UserCardProps> = (
     ...rest
   } = props
   const { accessToken } = useSynapseContext()
-  const [userProfile, setUserProfile] = useState(initialProfile)
   const [principalId, setPrincipalId] = useState(ownerId)
   const [isLoading, setIsLoading] = useState(true)
-  const [preSignedURL, setPresignedUrl] = useState<string | undefined>(
-    initialPreSignedURL,
-  )
+  const [fetchedPresignedUrl, setFetchedPresignedUrl] = useState<
+    string | null | undefined
+  >(undefined)
 
-  const { data: profileAndImage } = useGetUserProfileWithProfilePic(
-    principalId!,
-    {
-      enabled: !!principalId,
-    },
-  )
+  const { data: fetchedProfile } = useGetUserProfile(principalId!, {
+    enabled: !!principalId && !initialProfile,
+  })
+
+  const userProfile = initialProfile ?? fetchedProfile
 
   useEffect(() => {
-    if (profileAndImage) {
-      if (!initialProfile) {
-        setUserProfile(profileAndImage.userProfile)
-      }
-      if (!initialPreSignedURL) {
-        setPresignedUrl(profileAndImage.preSignedURL)
-      }
+    let isCanceled = false
+    if (!initialPreSignedURL && (ownerId || userProfile?.ownerId)) {
+      SynapseClient.getProfilePicPreviewPresignedUrl(
+        (ownerId ?? userProfile?.ownerId)!,
+      ).then(result => {
+        if (!isCanceled) {
+          setFetchedPresignedUrl(result)
+        }
+      })
```

Comment:
If we weren't provided `initialPreSignedURL` and we have an ID, fetch the presigned URL for the preview from Synapse.

Reply:
Nice use of the cleanup function